src/ProcessManagerBundle/EventListener/ArtifactListener.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ProcessManagerBundle\EventListener;
+
+use Pimcore\Event\Model\AssetEvent;
+use Pimcore\File as FileHelper;
+use Pimcore\Model\Asset;
+use ProcessManagerBundle\Model\Process;
+use ProcessManagerBundle\Model\ProcessInterface;
+use ProcessManagerBundle\Monolog\ProcessLogEvent;
+
+/**
+ * Class ArtifactListener.
+ */
+class ArtifactListener
+{
+    private const METADATA_MARKER = 'process_manager.process';
+
+    /**
+     * @param ImportDefinitionEvent $event
+     */
+    public function onProcessLogEvent(ProcessLogEvent $event)
+    {
+        // receives it via the logger from self::onArtifactEvent()
+        $record = $event->getRecord();
+
+        if (false === array_key_exists('artifact', $record['context'])) {
+            return;
+        }
+        
+        /** @var ProcessInterface $process */
+        $process = $record['extra']['process'];
+
+        // ProcessInterface instance must have a relation to its ExecutableInterface instance
+        // TODO: need asset path in Executable config, hardcoded for now
+        $artifactAssetPath = '/exports';
+
+        $artifactPath = $record['context']['artifact'];
+        $fileHandle = fopen($artifactPath, 'rb', false, FileHelper::getContext());
+
+        $artifact = new Asset();
```

Comment:
Still don't agree with that, the asset should be created from the process that creates the Process (eg. Import Definitions). Not every process has a executable linked to it. Even Import Definitions logs to PM without an executable.